codegen/src/api/calls.rs

```diff
@@ -32,34 +32,47 @@ pub fn generate_calls(
 
     let mut struct_defs = super::generate_structs_from_variants(
         type_gen,
+        types_mod_ident,
         call_ty,
         |name| name.to_upper_camel_case().into(),
         "Call",
         crate_path,
         should_gen_docs,
     )?;
-    let (call_structs, call_fns): (Vec<_>, Vec<_>) = struct_defs
+
+    let result = struct_defs
         .iter_mut()
         .map(|(variant_name, struct_def)| {
-            let (call_fn_args, call_args): (Vec<_>, Vec<_>) = match struct_def.fields {
-                CompositeDefFields::Named(ref named_fields) => named_fields
-                    .iter()
-                    .map(|(name, field)| {
-                        let fn_arg_type = &field.type_path;
-                        let call_arg = if field.is_boxed() {
-                            quote! { #name: ::std::boxed::Box::new(#name) }
-                        } else {
-                            quote! { #name }
-                        };
-                        (quote!( #name: #fn_arg_type ), call_arg)
-                    })
-                    .unzip(),
+            let fn_name = format_ident!("{}", variant_name.to_snake_case());
+
+            let result: Vec<_> = match struct_def.fields {
+                CompositeDefFields::Named(ref named_fields) => {
+                    named_fields
+                        .iter()
+                        .map(|(name, field)| {
+                            let call_arg = if field.is_boxed() {
+                                quote! { #name: ::std::boxed::Box::new(#name) }
+                            } else {
+                                quote! { #name }
+                            };
+                            // One downside of the type alias system is that we generate one alias per argument.
+                            // Multiple arguments could potentially have the same type (ie fn call(u32, u32, u32)).
```

Comment:
That worried me for a sec, but actually all you mean is that we might have alias like

```rust
type Foo = core::primitive::u8;
type Bar = core::primitive::u8;
```

Which is fine I think :)